great_expectations/checkpoint/util.py

```diff
@@ -239,76 +272,247 @@ def get_runtime_batch_request(
-            and runtime_config_batch_request_val is not None
-        ):
+    for key, value in validation_batch_request.items():
+        substituted_value = substituted_runtime_batch_request.get(key)
+        if value is not None and substituted_value is not None:
             raise ge_exceptions.CheckpointError(
                 f'BatchRequest attribute "{key}" was specified in both validation and top-level CheckpointConfig.'
             )
-    runtime_batch_request_dict.update(runtime_config_batch_request)
-
-    if ge_cloud_mode and batch_request_class is RuntimeBatchRequest:
-        batch_identifiers = runtime_batch_request_dict.get("batch_identifiers", {})
-        if len(batch_identifiers.keys()) == 0:
-            batch_identifiers["timestamp"] = str(datetime.datetime.now())
-            runtime_batch_request_dict["batch_identifiers"] = batch_identifiers
 
+    # noinspection PyUnresolvedReferences
     filter_properties_dict(
-        properties=runtime_batch_request_dict,
+        properties=effective_batch_request,
         keep_fields=batch_request_class.field_names,
         clean_nulls=False,
         inplace=True,
     )
 
-    return batch_request_class(**runtime_batch_request_dict)
+    return batch_request_class(**effective_batch_request)
 
 
-def get_substituted_validation_dict(
-    substituted_runtime_config: CheckpointConfig, validation_dict: dict
-) -> dict:
-    substituted_validation_dict = {
-        "batch_request": get_runtime_batch_request(
-            substituted_runtime_config=substituted_runtime_config,
-            validation_batch_request=validation_dict.get("batch_request"),
-        ),
-        "expectation_suite_name": validation_dict.get("expectation_suite_name")
-        or substituted_runtime_config.expectation_suite_name,
-        "expectation_suite_ge_cloud_id": validation_dict.get(
+def substitute_template_config(source_config: dict, template_config: dict) -> dict:
+    if not (template_config and any(template_config.values())):
+        return source_config
+
+    dest_config: dict = copy.deepcopy(template_config)
+
+    # replace
+    if source_config.get("name") is not None:
+        dest_config["name"] = source_config["name"]
+    if source_config.get("module_name") is not None:
+        dest_config["module_name"] = source_config["module_name"]
+    if source_config.get("class_name") is not None:
+        dest_config["class_name"] = source_config["class_name"]
+    if source_config.get("run_name_template") is not None:
+        dest_config["run_name_template"] = source_config["run_name_template"]
+    if source_config.get("expectation_suite_name") is not None:
+        dest_config["expectation_suite_name"] = source_config["expectation_suite_name"]
+    if source_config.get("expectation_suite_ge_cloud_id") is not None:
+        dest_config["expectation_suite_ge_cloud_id"] = source_config[
             "expectation_suite_ge_cloud_id"
+        ]
```

Comment:
Could we refactor this?

```python
attrs = ["name", "module_name", ...]

for attr in attrs:
  dest_config[attr] = source_config[attr]

```

Reply:
+1

Reply:
@cdkini Potentially -- could you please put this idea into a separate JIRA ticket for us to follow up on?  Thanks!